Plugins~/Src/MeshSyncClientBlender/python/unity_mesh_sync_baking.py

```diff
@@ -0,0 +1,597 @@
+                bpy.ops.object.mode_set(mode='EDIT')
+
+            bpy.ops.mesh.select_mode(use_extend=False, use_expand=False, type='VERT')
+            bpy.ops.mesh.select_all(action='SELECT')
+            bpy.ops.mesh.select_linked(delimit={'SEAM'})
+            bpy.ops.uv.smart_project(island_margin=0.01, scale_to_bounds=True)
+            bpy.ops.uv.pack_islands(rotate=True, margin=0.001)
+
+    def getNodeYLocation(self, node):
+        location = node.location[1]
+        if node.parent is not None:
+            location += self.getNodeYLocation(node.parent)
+        return location
+
+    def prepareBake(self, context, obj, bsdf, mat):
+        if context.object is not None:
+            bpy.ops.object.mode_set(mode='OBJECT')
+        # bpy.ops.object.select_all(view_3d, action='DESELECT')
+        for ob in context.selected_objects:
+            ob.select_set(False)
+        obj.select_set(True)
+        bpy.context.view_layer.objects.active = obj
+
+        # Make material copy if this is not a copy already:
+        if ORIGINAL_MATERIAL not in mat:
+            # Use existing copy if there is one:
+            matCopyName = f"{mat.name}_{obj.name}_baked"
+            matCopyIndex = bpy.data.materials.find(matCopyName)
+            if matCopyIndex >= 0:
+                matCopy = bpy.data.materials[matCopyIndex]
+            else:
+                mat.use_fake_user = True    # Make sure this does not get deleted when it's not referenced anymore
+                matCopy = mat.copy()
+                matCopy[ORIGINAL_MATERIAL] = mat.name
+                matCopy.name = f"{mat.name}_{obj.name}_baked"
+
+                print(f"Creating material copy '{mat.name}'->'{matCopy.name}'")
+
+                bakedBSDF = matCopy.node_tree.nodes.new(type='ShaderNodeBsdfPrincipled')
+
+                # Find lowest node in the tree and put the baked bsdf under that:
+                minYLocation = bsdf.location[1]
+                for node in matCopy.node_tree.nodes:
+                    minYLocation = min(minYLocation, self.getNodeYLocation(node))
+
+                bakedBSDF.location = (bsdf.location[0], minYLocation - 1000)
+                # Give bsdf a name and set its name on the material, so we can find it again:
+                bakedBSDF.name = BAKED_MATERIAL_SHADER
+                matCopy[BAKED_MATERIAL_SHADER] = bakedBSDF.name
+
+                matIndex = obj.material_slots.find(mat.name)
+                obj.material_slots[matIndex].material = matCopy
+            # obj.data.materials.pop(index=obj.data.materials.find(mat.name))
+            # obj.data.materials.append(matCopy)
+
+            mat = matCopy
+
+        self.ensureUVs(context, obj)
+
+        scene = context.scene
+
+        if obj.mode == 'EDIT':
+            bpy.ops.object.mode_set(mode='OBJECT')
+
+        scene.render.engine = "CYCLES"
+        scene.cycles.device = "GPU"
+        scene.cycles.samples = 10       # TODO: This could be a setting for the user to change
```

Comment:
I agree with the TODO comment. Also, does cycles support all bsdf that Eevee supports?

Reply:
As far as I know, cycles supports everything eevee supports and more.